src/main/java/io/reactivex/Flowable.java

```diff
@@ -400,7 +398,27 @@ public static int bufferSize() {
         if (sources.length == 1) {
             return fromPublisher(sources[0]);
         }
-        return fromArray(sources).concatMap((Function)Functions.identity());
+        return new FlowableConcatArray<T>(sources, false);
+    }
+
+    /**
+     * Concatenates a variable number of Observable sources and delays errors from any of them
+     * till all terminate.
+     * @param sources the array of sources
+     * @param <T> the common base value type
+     * @return the new Flowable instance
+     * @throws NullPointerException if sources is null
+     */
+    @BackpressureSupport(BackpressureKind.FULL)
+    @SchedulerSupport(SchedulerSupport.NONE)
+    public static <T> Flowable<T> concatArrayDelayError(Publisher<? extends T>... sources) {
+        if (sources.length == 0) {
+            return empty();
+        } else
```

Comment:
But how's that any different from the overwhelmingly more common `else if` on one line?